src/node_buffer.cc

```diff
@@ -1072,6 +1072,42 @@ void IndexOfNumber(const FunctionCallbackInfo<Value>& args) {
                                 : -1);
 }
 
+void SwapShort(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  THROW_AND_RETURN_UNLESS_BUFFER(env, args.This());
+  SPREAD_ARG(args.This(), ts_obj);
+
+  if (ts_obj_length % 2 != 0)
+    return env->ThrowRangeError(
+      "Buffer length must be a multiple of 16-bits");
+
+  for (size_t i = 0; i < ts_obj_length; i += 2) {
+    const uint8_t lo = static_cast<uint8_t>(ts_obj_data[i + 0]);
+    const uint8_t hi = static_cast<uint8_t>(ts_obj_data[i + 1]);
+    ts_obj_data[i + 0] = hi;
+    ts_obj_data[i + 1] = lo;
+  }
+  args.GetReturnValue().Set(args.This());
+}
+
+void SwapLong(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  THROW_AND_RETURN_UNLESS_BUFFER(env, args.This());
+  SPREAD_ARG(args.This(), ts_obj);
+
+  if (ts_obj_length % 4 != 0)
+    return env->ThrowRangeError(
+      "Buffer length must be a multiple of 32-bits");
+
+  uint32_t* data = reinterpret_cast<uint32_t*>(ts_obj_data);
```

Comment:
@trevnorris ... on a slice, the length check immediately above should be enough to catch the alignment, should it not? e.g. `Buffer([0x1,0x2,0x3,0x4,0x5]).slice(1,5).swap32()` works just fine... and `Buffer([0x1,0x2,0x3,0x4,0x5]).slice(1,4).swap32()` throws as it would be expected to. Or am I missing something?